scalableminds/webknossos
# Tiff Export Job

Lets the user download a zipfile with tiff files, exported from wkw on disk.

- New api routes `api/jobs/run/tiffExport/:organizationName/:dataSetName/:layerName/:bbox` (with bbox being in format x,y,z,w,h,d)
- `api/jobs/downloadExport/:organizationName/:exportFileName` (jobs json may now include exportFileName, I already adapted api flow types)

### Todo: 
- [x] create new task in webknossos-worker (see https://github.com/scalableminds/webknossos-worker/pull/5 )
- [x] create api route to call that job
- [x] expose exported files to download
- [x] adapt jobs list to show this download link
- [x] Create UI that calls the run api
- [x] Create follow-up issue for expiry + cleanup job → #5202
- [x] Create follow-up issue for exporting volume annotations → #5203

### Steps to test:
- Set up https://github.com/scalableminds/webknossos-worker/pull/5 (Will be merged into master there once this is approved)
- set `features.jobsEnabled = true` in `application.conf`
- Create annotation, create user bounding box
- In user bounding box list, you should see download icons, opens modal
- select layer, see that job is running in jobs page
- when complete, download zip, should contain selected data range

### Issues:
- fixes #4911 

------
